model/event/event_finder.go

```diff
@@ -61,6 +63,36 @@ func CountProcessEvents(taskID string) (int, error) {
 	return db.CountQ(TaskLogCollection, db.Query(filter))
 }
 
+func FindLastProcessedEvent() (*EventLogEntry, error) {
+	q := db.Query(bson.M{
+		processedAtKey: bson.M{
+			"$exists": true,
+		},
+	}).Sort([]string{"-" + processedAtKey}).Limit(1)
```

Comment:
I think the limit 1 is belt and suspenders with the FinOne

Reply:
removed.